src/ucp/core/ucp_worker.c

```diff
@@ -2665,9 +2683,20 @@ void ucp_worker_destroy(ucp_worker_h worker)
 
     UCS_ASYNC_BLOCK(&worker->async);
     uct_worker_progress_unregister_safe(worker->uct, &worker->keepalive.cb_id);
+
+    /* Destroying UCP EP in ucp_ep_disconnected() could start UCT EP discarding
+     * operations.
+     * Do cleanup of discarding functionality two times:
+     * - before destroying of all UCP EP in order to destroy all remaining UCP
+     *   EPs (they are destroyed during discarding operation completion for all
+     *   UCT EPs)
+     * - after destroying of all UCP EP in order to destroy all remaining UCP
+     *   EPs for whih */
```

Comment:
what if we cleanup discard only the 2nd time (after calling XX_destroy_eps)?
also comment seems incomplete

Reply:
then it would be problematic, since `ucp_ep_delete` would be called twice for EPs which was destroyed and we scheduled discard operations for them.
fixed and extended the comment